### PR TITLE
feat: ability to reorder fields in multiple select in Routing Forms

### LINKED ISSUE #10918: [CAL-2385] Ability to Reorder Fields Easier in Multiple Select in Routing Forms
Currently, to reorder fields, you need to copy and delete each field to reorder them, it would be great to have a toggle like how we can reorder the routes

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2385](https://linear.app/calcom/issue/CAL-2385/ability-to-reorder-fields-easier-in-multiple-select-in-routing-forms)</sub>

### PR DESCRIPTION
**Pull request by G3root**
## What does this PR do?

Fixes #10918

[Screencast 2023-08-24 17:06:41.webm](https://github.com/calcom/cal.com/assets/84864519/30f35bfb-3f58-4c6f-8a69-92df44e4ef5a)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Review by Udit-takkar — COMMENTED** [review]
@G3root Works fine. can you also add animations like we do in /event-types?

https://github.com/calcom/cal.com/assets/53316345/cb8e05a9-5089-4ed1-8206-2500b44dd631

**Comment by G3root** [comment]
@Udit-takkar done 

[Screencast 2023-09-26 15:44:57.webm](https://github.com/calcom/cal.com/assets/84864519/7618b3e7-c188-4738-abb9-7cfa1b21f01f)

File: packages/app-store/routing-forms/pages/form-edit/[...appPages].tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ import SingleForm, {
 
 export { getServerSideProps };
 type HookForm = UseFormReturn<RoutingFormWithResponseCount>;
-type SelectOption = { placeholder: string; value: string };
+type SelectOption = { placeholder: string; value: string; id: string };
```
Comment (G3root) [inline packages/app-store/routing-forms/pages/form-edit/[...appPages].tsx:32]: use animate need a consistent key to work

**Review by Udit-takkar — APPROVED** [review]
LGTM

**Pull request merged**